iwyu.cc

```diff
@@ -3708,7 +3709,19 @@ class IwyuAstConsumer
     // though, because that will drag in every overload even if we're
     // only using one.  Instead, we keep track of the using decl and
     // mark it as touched when something actually uses it.
-    preprocessor_info().FileInfoFor(CurrentFileEntry())->AddUsingDecl(decl);
+    IwyuFileInfo* file_info =
+        preprocessor_info().FileInfoFor(CurrentFileEntry());
+    if (file_info) {
+      file_info->AddUsingDecl(decl);
+    } else {
+      // The precomiled header file will not (yet) be found in the
```

Comment:
The formatting is a little weird here, and the comment could be more to the point. I suggest:

```diff
@@ -3708,7 +3708,19 @@ class IwyuAstConsumer
     // though, because that will drag in every overload even if we're
     // only using one.  Instead, we keep track of the using decl and
     // mark it as touched when something actually uses it.
-    preprocessor_info().FileInfoFor(CurrentFileEntry())->AddUsingDecl(decl);
+    IwyuFileInfo* file_info =
+        preprocessor_info().FileInfoFor(CurrentFileEntry());
+    if (file_info) {
+      file_info->AddUsingDecl(decl);
+    } else {
+      // For using declarations in a PCH, the preprocessor won't have any
+      // location information. As far as we know, that's the only time the
+      // file-info will be null, so assert that we have a PCH on the
+      // command-line.
+      const string& pch_include =
+          compiler()->getInvocation().getPreprocessorOpts().ImplicitPCHInclude;
+      CHECK_(!pch_include.empty());
+    }
 
     if (CanIgnoreCurrentASTNode())  return true;
```